### PR TITLE
🎉 Survey Monkey: Improve 'check' using '/users/me' API call

### LINKED ISSUE #7817: Bug: Error message should be more readable
Update the Error Message to more readable for user.
image

## Enviroment
- **Airbyte version**:  0.30.31-alpha
- **OS Version / Instance**: Windows 10
- **Deployment**: Docker deploy env
- **Source Connector and version**: Survey Monkey 0.1.3
- **Severity**:  High
- **Step where error happened**:  Setup source

## Current Behavior
Start Date: 2021-11-01T00:00:00Z
On *Setup source* button press the Error message is "The connection tests failed. 'StopIteration()'"

## Expected Behavior
Put the Error message which explains to user that the date they selected is not valid.

## Logs
Check connection:
{status: "failed", message: "'StopIteration()'",…}
jobInfo: {id: "c4d1a8c3-fa7b-4cb8-ae0b-618f42edfbf2", configType: "check_connection_source",…}
configId: "Optional[badc5925-0485-42be-8caa-b34096cb71b5]"
configType: "check_connection_source"
createdAt: 1636553557791
endedAt: 1636553559615
id: "c4d1a8c3-fa7b-4cb8-ae0b-618f42edfbf2"
logs: {logLines: [,…]}
logLines: [,…]
0: "2021-11-10 14:12:37 INFO () TemporalAttemptExecution(get):116 - Executing worker wrapper. Airbyte version: dev"
1: "2021-11-10 14:12:37 INFO () LineGobbler(voidCall):82 - Checking if airbyte/source-surveymonkey:0.1.3 exists..."
2: "2021-11-10 14:12:37 INFO () LineGobbler(voidCall):82 - airbyte/source-surveymonkey:0.1.3 was found locally."
3: "2021-11-10 14:12:37 INFO () DockerProcessFactory(create):127 - Preparing command: docker run --rm --init -i -v airbyte_workspace:/data -v /tmp/airbyte_local:/local -w /data/f8df86ef-9306-43d5-bcaf-33e855c152f6/0 --network host --log-driver none airbyte/source-surveymonkey:0.1.3 check --config source_config.json"
4: "2021-11-10 14:12:39 ERROR () DefaultAirbyteStreamFactory(internalLog):95 - Check failed"
5: "2021-11-10 14:12:39 INFO () TemporalAttemptExecution(get):137 - Stopping cancellation check scheduling..."
succeeded: true
message: "'StopIteration()'"
status: "failed"

## Steps to Reproduce
1. Enter Source Type = Survey Monkey
2. Enter Start Date = 2021-11-01T00:00:00Z
3. Press Authenticate your Survey Monkey account button
4. Press Setup source button

### PR DESCRIPTION
**Pull request by grubberr**
Signed-off-by: Sergey Chvalyuk <grubberr@gmail.com>

## What
In current implementation of "check" command connector reads "Surveys" stream which can be empty.
If stream is empty we got error but in really - connection is OK.

## How
In new implementation, we call API endpoint `https://api.surveymonkey.com/v3/users/me` to make sure we have valid credentials and also as bonus check that we have all needed oAuth scopes.

## Recommended reading order
1. `x.java`
2. `y.python`

## Pre-merge Checklist
Expand the relevant checklist and delete the others. 

<details><summary> <strong> Updating a connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the new connector version is published, connector version bumped in the seed directory as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</p>
</details>

### CODE REVIEW

**Review by midavadim — APPROVED** [review]

**Review by lazebnyi — APPROVED** [review]

File: airbyte-integrations/connectors/source-surveymonkey/source_surveymonkey/source.py [hunk 1]
```diff
@@ -36,3 +53,10 @@ def streams(self, config: Mapping[str, Any]) -> List[Stream]:
     def get_authenticator(config: Mapping[str, Any]):
         token = config["access_token"]
         return TokenAuthenticator(token=token)
+
+    @classmethod
+    def _check_scopes(cls, response_json):
+        granted_scopes = response_json["scopes"]["granted"]
+        missed_scopes = cls.SCOPES - set(granted_scopes)
+        if missed_scopes:
+            raise Exception("missed required scopes: " + ", ".join(missed_scopes))
```
Comment (antixar) [inline airbyte-integrations/connectors/source-surveymonkey/source_surveymonkey/source.py:62]: Please use a direct error handling here:
return False, "missed required scopes: " + ", ".join(missed_scopes)"

Comment (grubberr) [reply airbyte-integrations/connectors/source-surveymonkey/source_surveymonkey/source.py]: Done!

**Review by antixar — APPROVED** [review]
Looks good! @grubberr , please merge and publish your fix because your solution is simple and obvious.

**Pull request merged**